Analiza plimorfizmu genu Vdr wariantów foc1 i bsm1
Dane: 2 grypy, jedna z nowotowrem nerki druga zdrowa,
analiza czestości wystepowania polimofrizmów.

Gen VDR (ang. Vitamin D Receptor) koduje białko zwane receptorem witaminy D (VDR). Receptor witaminy D jest białkiem, które odgrywa kluczową rolę w regulacji odpowiedzi komórkowej na witaminę D.

Wskazuje się związek polimorfizmów genu VDR z ryzykiem wystąpienia raka sutka (Bsm1, Fok1), raka prostaty (Fok1) i czerniaka złośliwego (Fok1). Największe znaczenie w rokowaniu obserwuje się u pacjentów z rakiem prostaty (Fok1), rakiem piersi (Bsm1, Taq1), czerniakiem złośliwym o (Bsm1) i rakiem nerkowokomórkowym (Taq1). Istotnym jest rozpoznanie, opisanie i skorelowanie występowania genetycznego polimorfizmów genu VDR, które pozwolą na wczesne rozpoznanie bądź zapobieganie wystąpienia skorelowanych z nimi jednostek chorobowych.

In [ ]:
install.packages('readxl')
library(readxl)


Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
dane <- read_excel("/dane_projektt.xls")


In [ ]:
dim(dane)

[1] 646   3

This second cell will be in a separate sandboxed iframe.


In [ ]:
colnames(dane) <- c('Grupa','VDR','BSM')

In [ ]:
table(dane$Grupa)
table(dane$VDR)
table(dane$BSM)


 Cancer Control 
    276     369 


  1  AA  AG  GG 
  1 157 264 224 


 CC  CT  TC  TT 
202 142 154  84 

In [ ]:
dane<-dane[dane$Grupa != '',]
dim(dane)

[1] 646   3

In [ ]:
table(dane$VDR)
table(dane$BSM)


 AA  AG  GG 
157 264 224 


 CC  CT  TC  TT 
202 142 154  84 

In [ ]:
BSMdata<-dane[dane$BSM != '',]
table(BSMdata$BSM)


 CC  CT  TC  TT 
202 142 154  84 

In [ ]:
dim(VDRcase<-dane[dane$Grupa == 'Cancer',])
dim(VDRcontrol<-dane[dane$Grupa == 'Control',])
dim(BSMcase<-BSMdata[BSMdata$Grupa == 'Cancer',])
dim(BSMcontrol<-BSMdata[BSMdata$Grupa == 'Control',])

[1] 277   3

[1] 370   3

[1] 340   3

[1] 370   3

Sprawdzmy czy populacja jest w staie równowagi HW

1) Cancer

Liczebość obserwacji

In [ ]:
table(VDRcase$VDR)


AA AG GG 
92 94 90 

In [ ]:
N<-sum(table(VDRcase$VDR))
AA<-92
AG<-94
GG<-90

Liczembość alleli

In [ ]:
(nA<-2*AA+AG)
(nG<-2*GG+AG)

[1] 278

[1] 274

In [ ]:
(p<-nA/(2*N))
(q<-nG/(2*N))
p+q

[1] 0.5036232

[1] 0.4963768

[1] 1

oczekiwana liczbośc genów

In [ ]:
eAA<-p^2*N
eAG<-2*p*q*N
eGG<-q^2*N

Statystyka chi-kwadrat

In [ ]:
VDRc<-((92-eAA)^2/eAA)+((94-eAG)^2/eAG)+((90-eGG)^2/eGG) #suma kwadratów różnic
alfa=0.05 #poziom istotonscio
df=3-1  #stopnie sfobody
VDRc

[1] 28.05168

In [ ]:
qchisq(1-alfa,df)

[1] 5.991465

In [ ]:
p_value <- 1 - pchisq(VDRc, df)
alfa <- 0.05

p_value

[1] 8.103189e-07

In [ ]:
chisq.test(c(92,94,90), p=c(eAA/N,eAG/N,eGG/N))$p.value

[1] 8.103189e-07

H0: Populacja osób chorych na raka znajduje się w stanie  równowagi Hardy'ego-Weinberga.

h1: Populacja osób chorych na raka nie znajduje się w stanie  równowagi Hardy'ego-Weinberga.

x^2 = 28.05 > 5,99 = x^2(0,05)

p.value = 8.103189e-07 < 0.05

Brak podstaw do przyjęcia hipotezy zerowej, subpopulacja nie znajduje się w stanie równowagi Hardy'ego-Weinberga.

2) Contreol

In [ ]:
table(VDRcontrol$VDR)


 AA  AG  GG 
 65 170 134 

In [ ]:
N2<-sum(table(VDRcontrol$VDR))
(nAvdr<-2*65+170)
(nGvdr<-2*134+170)

(pvdr<-nAvdr/(2*N2))
(qvdr<-nGvdr/(2*N2))

[1] 300

[1] 438

[1] 0.4065041

[1] 0.5934959

In [ ]:
eAAvdr<-pvdr^2*N2
eAGvdr<-2*pvdr*qvdr*N2
eGGvdr<-qvdr^2*N2

VDRcc<-((65-eAAvdr)^2/eAAvdr)+((170-eAGvdr)^2/eAGvdr)+((134-eGGvdr)^2/eGGvdr)
alfa=0.05
dff=3-1
VDRcc
qchisq(1-alfa,df)

[1] 0.7540646

[1] 5.991465

In [ ]:
p_value <- 1 - pchisq(VDRcc, dff)
alfa <- 0.05

p_value

[1] 0.6858939

H0: Populacja osób chorych na raka znajduje się w stanie równowagi Hardy'ego-Weinberga.

H1: Populacja osób chorych na raka nie znajduje się w stanie równowagi Hardy'ego-Weinberga.

x^2 = 0,75 < 5,99 =x^2(0,05)

p.value = 0.6858939 > 0,05

Brak podstaw do odrzucenia hipotezy zerowej, subpoplulacja znajduje się w stanie równowagi Hardy'ego-Weinberga.

Wariant BSM

1) Cancer

In [ ]:
(t3<-table(BSMcase$BSM))
N3<-sum(t3)


CC CT TC TT 
96 71 69 40 

In [ ]:
pC<-(2*96+71+69)/(2*N3)
pT<-(2*40+71+69)/(2*N3)

eCCa <- pC^2*N3
eCTa <- 2*pC*pT*N3
eTTa <- pT^2*N3

BSMc<-((96-eCCa)^2/eCCa)+(((71+69)-eCTa)^2/eCTa)+((40-eTTa)^2/eTTa)
alfa=0.05
df=3-1
BSMc
qchisq(1-alfa,df)

[1] 0.9300778

[1] 5.991465

In [ ]:
p_value <- 1 - pchisq(BSMc, df)
alfa <- 0.05

p_value

[1] 0.6281107

In [ ]:
chisq.test(c(96,71+69,40), p=c(eCCa/N3,eCTa/N3,eTTa/N3))


	Chi-squared test for given probabilities

data:  c(96, 71 + 69, 40)
X-squared = 0.93008, df = 2, p-value = 0.6281


H0: Populacja osób chorych na raka znajduje się w stanie równowagi Hardy'ego-Weinberga.

h1: Populacja osób chorych na raka nie znajduje się w stanie równowagi Hardy'ego-Weinberga.

x^2 = 0,93 < 5,99 =x^2(0,05)

p.value = 0.6281 > 0,05

Brak podstaw do odrzucenia hipotezy zerowej, subpoplulacja znajduje się w stanie równowagi Hardy'ego-Weinberga.

2) Control

In [ ]:
(t4<-table(BSMcontrol$BSM))
N4<-sum(t4)


 CC  CT  TC  TT 
106  71  85  44 

In [ ]:
pCb<-(2*106+71+85)/(2*N4)
qTb<-(2*44+71+85)/(2*N4)

eCCb <- pCb^2*N4
eCTb <- 2*pCb*qTb*N4
eTTb <- qTb^2*N4

BSMcc<-((106-eCCb)^2/eCCb)+(((71+85)-eCTb)^2/eCTb)+((44-eTTb)^2/eTTb)
alfa=0.05
df=3-1
BSMcc
qchisq(1-alfa,df)

[1] 1.224455

[1] 5.991465

In [ ]:
p_value <- 1 - pchisq(BSMcc, df)
alfa <- 0.05

p_value

[1] 0.5421419

In [ ]:
chisq.test(c(106,71+85,44), p=c(eCCa/N3,eCTa/N3,eTTa/N3))

H0: Populacja osób chorych na raka znajduje się w stanie równowagi Hardy'ego-Weinberga.

h1: Populacja osób chorych na raka nie znajduje się w stanie równowagi Hardy'ego-Weinberga.

x^2 = 1,22 < 5,99 =x^2(0,05)

p.value = 0.5421419 > 0,05

Brak podstaw do odrzucenia hipotezy zerowej, subpoplulacja znajduje się w stanie równowagi Hardy'ego-Weinberga.

Róznice w występowaniu cześci genotypów

BSM

In [ ]:
table(BSMcontrol$BSM)
table(BSMcase$BSM)


 CC  CT  TC  TT 
106  71  85  44 


CC CT TC TT 
96 71 69 40 

In [ ]:
m1=matrix(c(106,71,85,44,96,71,69,40),ncol=2)
colnames(m1)=c("control","case")
rownames(m1)=c("CC","CT","TC","TT")
m1

control case
CC 106     96  
CT  71     71  
TC  85     69  
TT  44     40

In [ ]:
m1[,1]<-m1[,1]/sum(m1[,1])
m1[,2]<-m1[,2]/sum(m1[,2])
m1

control   case     
CC 0.3464052 0.3478261
CT 0.2320261 0.2572464
TC 0.2777778 0.2500000
TT 0.1437908 0.1449275

In [ ]:
stat1<-((m1[1,1]-m1[1,2])^2/m1[1,2])+((m1[2,1]-m1[2,2])^2/m1[2,2])+((m1[3,1]-m1[3,2])^2/m1[3,2])+((m1[4,1]-m1[4,2])^2/m1[4,2])
stat1

[1] 0.005573711

In [ ]:
alfa=0.05
df=4-1
qchisq(1-alfa,df)

[1] 7.814728

In [ ]:
chisq.test(m1[,1], p=m1[,2])

Warning message in chisq.test(m1[, 1], p = m1[, 2]):
“Chi-squared approximation may be incorrect”



	Chi-squared test for given probabilities

data:  m1[, 1]
X-squared = 0.0055737, df = 3, p-value = 0.9999


H0: Brak różnic biedzy subpopulacjami

H1: Wystepują różnice miedzy subpopulacjami


x^2 = 0.0055737 < 7.814728 =x^2(0,05)

p.value = 1 > 0,05

Brak podstaw do odrzucenia hipotezy zerowej, nie ma różnic w występowaniu genotypów miedzy subpopulacjiami.

In [ ]:
table(VDRcontrol$VDR)
table(VDRcase$VDR)


 AA  AG  GG 
 65 170 134 


AA AG GG 
92 94 90 

In [ ]:
m2=matrix(c(65, 170, 134, 92, 94, 90),ncol=2)
colnames(m2)=c("control","case")
rownames(m2)=c("AA","AG","GG")
m2

control case
AA  65     92  
AG 170     94  
GG 134     90

In [ ]:
m2[,1]<-m2[,1]/sum(m2[,1])
m2[,2]<-m2[,2]/sum(m2[,2])
m2

control   case     
AA 0.1761518 0.3333333
AG 0.4607046 0.3405797
GG 0.3631436 0.3260870

In [ ]:
stat2<-((0.1761518-0.3333333)^2/0.3333333) + ((0.4607046-0.3405797)^2/0.3405797) + ((0.3631436-0.3260870)^2/0.3260870)
stat2

[1] 0.1206981

In [ ]:
alfa=0.05
df=3-1
qchisq(1-alfa,df)

[1] 5.991465

In [ ]:
chisq.test(m2[,1], p=m2[,2])

Warning message in chisq.test(m2[, 1], p = m2[, 2]):
“Chi-squared approximation may be incorrect”



	Chi-squared test for given probabilities

data:  m2[, 1]
X-squared = 0.1207, df = 2, p-value = 0.9414


H0: Brak różnic biedzy subpopulacjami

H1: Wystepują różnice miedzy subpopulacjami


x^2 = 0.12 < 5.99 =x^2(0,05)

p.value = 0.94 > 0.05

Brak podstaw do odrzucenia hipotezy zerowej, nie ma różnic w występowaniu genotypów miedzy subpopulacjiami.


Iloraz szans(Odss ratio)

Dla wariantu VDR.Floki w grupie chorych na raka.

In [ ]:
nA_VDR_case <- 92
nG_VDR_case <- 94 + 90

odds_ratio_VDR_case <- nA_VDR_case / nG_VDR_case
cat("Odds Ratio dla wariantu VDR.Floki w grupie chorych na raka:", odds_ratio_VDR_case, "\n")

Odds Ratio dla wariantu VDR.Floki w grupie chorych na raka: 0.5 


In [ ]:
nA_VDR_control <- 65
nG_VDR_control <- 170 + 134

odds_ratio_VDR_control <- nA_VDR_control / nG_VDR_control
cat("Odds Ratio dla wariantu VDR.Floki w grupie kontrolnej:", odds_ratio_VDR_control, "\n")

Odds Ratio dla wariantu VDR.Floki w grupie kontrolnej: 0.2138158 


In [ ]:
nC_BSM_case <- 96
nT_BSM_case <- 71 + 69

odds_ratio_BSM_case <- nC_BSM_case / nT_BSM_case
cat("Odds Ratio dla wariantu BSM w grupie chorych na raka:", odds_ratio_BSM_case, "\n")

Odds Ratio dla wariantu BSM w grupie chorych na raka: 0.6857143 


In [ ]:
nC_BSM_control <- 106
nT_BSM_control <- 71 + 85

odds_ratio_BSM_control <- nC_BSM_control / nT_BSM_control
cat("Odds Ratio dla wariantu BSM w grupie kontrolnej:", odds_ratio_BSM_control, "\n")

Odds Ratio dla wariantu BSM w grupie kontrolnej: 0.6794872 
